app/models/piece.rb

```diff
@@ -30,7 +30,7 @@ def square_occupied?(new_x, new_y)
     true
   end
 
-  def Piece.get_piece_at_coor(x, y)
+  def self.get_piece_at_coor(x, y)
     Piece.find_by(x_position: x, y_position: y)
   end
```

Comment:
Suggestion: This method is probably not needed. Both are the same length.

I'd stick with using `Piece.find_by(x_position: x, y_position: y)` everywhere.